csharp/src/Ice/Communicator.cs

```diff
@@ -243,39 +246,33 @@ internal CancellationToken CancellationToken
         private volatile ImmutableList<DispatchInterceptor> _defaultDispatchInterceptors =
             ImmutableList<DispatchInterceptor>.Empty;
         private Task? _destroyTask;
-        private static readonly Dictionary<string, Assembly> _loadedAssemblies = new Dictionary<string, Assembly>();
-        private readonly ConcurrentDictionary<ILocatorPrx, LocatorInfo> _locatorInfoMap =
-            new ConcurrentDictionary<ILocatorPrx, LocatorInfo>();
-
-        private volatile ILogger _logger;
-
-        private readonly object _mutex = new object();
-        private static bool _oneOffDone;
-        private static bool _printProcessIdDone;
-        private readonly ConcurrentDictionary<
-            string, Func<string?, RemoteExceptionOrigin?, RemoteException>?> _remoteExceptionFactoryCache = new();
-        private int _retryBufferSize;
-        private readonly ConcurrentDictionary<IRouterPrx, RouterInfo> _routerInfoTable =
-            new ConcurrentDictionary<IRouterPrx, RouterInfo>();
-        private readonly Dictionary<Transport, BufWarnSizeInfo> _setBufWarnSize =
-            new Dictionary<Transport, BufWarnSizeInfo>();
-        private bool _shutdown;
-        private TaskCompletionSource<object?>? _waitForShutdownCompletionSource;
 
         private readonly IDictionary<Transport, Ice1EndpointFactory> _ice1TransportRegistry =
             new ConcurrentDictionary<Transport, Ice1EndpointFactory>();
 
-        private readonly IDictionary<Transport, (Ice2EndpointFactory, Ice2EndpointParser)> _ice2TransportRegistry =
-            new ConcurrentDictionary<Transport, (Ice2EndpointFactory, Ice2EndpointParser)>();
-
         private readonly IDictionary<string, (Ice1EndpointParser, Transport)> _ice1TransportNameRegistry =
             new ConcurrentDictionary<string, (Ice1EndpointParser, Transport)>();
 
+        private readonly IDictionary<Transport, (Ice2EndpointFactory, Ice2EndpointParser)> _ice2TransportRegistry =
```

Comment:
Just declare the dictionary as a ConcurrentDictionary instead of IDictionary to be consistent with _locatorInfoMap and less verbose?

Reply:
I tried but IDictionary is better. We're using Add - which has the desired semantics - and ConcurrentDictionary does not provide Add directly.